File: ruby/easy/range_addition_ii.rb
# @param {Integer} m
# @param {Integer} n
# @param {Integer[][]} ops
# @return {Integer}
def max_count(m, n, ops)
    min_r = m
    min_c = n
    ops.each do |op|
        min_r = [min_r, op[0]].min
        min_c = [min_c, op[1]].min
    end
    return min_r * min_c
end

File: javascript/easy/rangeAdditionII.js
/**
 * @param {number} m
 * @param {number} n
 * @param {number[][]} ops
 * @return {number}
 */
var maxCount = function(m, n, ops) {
    let minR = m, minC = n;
    for(let op of ops) {
        if(op[0] < minR) {
            minR = op[0];    
        }
        
        if(op[1] < minC) {
            minC = op[1];
        }
    }
    return minR * minC;
};
